app/controllers/sessions_controller/lti_dependency.rb

```diff
@@ -94,7 +94,7 @@ def handle_lti_launch_error(err)
     error_msg = err.message
     message = err.lti_message
 
-    if message && message.launch_presentation_return_url
+    if message.try(:launch_presentation_return_url)
```

Comment:
Non blocking: does this do the same?

```suggestion
    if message&.launch_presentation_return_url
```

Reply:
I wish it did, but within the `IMS::LTI` gem our LTI code is built on top of, there’s a few different subclasses of `Message`, though for some reason they’ve decided to only let valid subclasses respond to `launch_presentation_return_url`, so in the case an LTI Launch Message is not well formed and it’s treated as a generic `Message` instance  it actually won’t respond to `launch_presentation_return_url`.